src/Bigcommerce/Api/Client.php

```diff
@@ -9,900 +9,1014 @@
-	 */
-	public static function getRequestLogs()
-	{
-		return self::getCollection('/requestlogs', 'RequestLog');
-	}
-
-	public static function getStore()
-	{
-		$response = self::connection()->get(self::$api_path . '/store');
-		return $response;
-	}
-
-	/**
-	 * The number of requests remaining at the current time. Based on the
-	 * last request that was fetched within the current script. If no
-	 * requests have been made, pings the time endpoint to get the value.
-	 *
-	 * @return int
-	 */
-	public static function getRequestsRemaining()
-	{
-		$limit = self::connection()->getHeader('X-BC-ApiLimit-Remaining');
-
-		if (!$limit) {
-			$result = self::getTime();
-
-			if (!$result) return false;
-
-			$limit = self::connection()->getHeader('X-BC-ApiLimit-Remaining');
-		}
-
-		return intval($limit);
-	}
-
-
-
+    static private $store_url;
+    static private $username;
+    static private $api_key;
+    static private $connection;
+    static private $resource;
+    static private $path_prefix = '/api/v2';
+
+    /**
+     * Full URL path to the configured store API.
+     *
+     * @var string
+     */
+    static public $api_path;
+
+    /**
+     * Configure the API client with the required credentials.
+     *
+     * Requires a settings array to be passed in with the following keys:
+     *
+     * - store_url
+     * - username
+     * - api_key
+     *
+     * @param array $settings
+     * @throws \Exception
+     */
+    public static function configure($settings)
+    {
+        if (!isset($settings['store_url'])) {
+            throw new Exception("'store_url' must be provided");
+        }
+
+        if (!isset($settings['username'])) {
+            throw new Exception("'username' must be provided");
+        }
+
+        if (!isset($settings['api_key'])) {
+            throw new Exception("'api_key' must be provided");
+        }
+
+        self::$username = $settings['username'];
+        self::$api_key = $settings['api_key'];
+        self::$store_url = rtrim($settings['store_url'], '/');
+        self::$api_path = self::$store_url . self::$path_prefix;
+        self::$connection = false;
+    }
+
+    /**
+     * Configure the API client to throw exceptions when HTTP errors occur.
+     *
+     * Note that network faults will always cause an exception to be thrown.
+     */
+    public static function failOnError($option = true)
+    {
+        self::connection()->failOnError($option);
+    }
+
+    /**
+     * Return XML strings from the API instead of building objects.
+     */
+    public static function useXml()
+    {
+        self::connection()->useXml();
+    }
+
+    /**
+     * Return JSON objects from the API instead of XML Strings.
+     * This is the default behavior.
+     */
+    public static function useJson()
+    {
+        self::connection()->useXml(false);
+    }
+
+    /**
+     * Switch SSL certificate verification on requests.
+     */
+    public static function verifyPeer($option = false)
```

Comment:
The only potential issue here is we are changing existing behavior. While this is going to be a new major release version, it still may be confusing.